pytorch_lightning/trainer/evaluation_loop.py

```diff
@@ -213,13 +224,14 @@ def evaluation_epoch_end(self, outputs: EPOCH_OUTPUT) -> None:
                 model._current_fx_name = 'validation_epoch_end'
                 model.validation_epoch_end(outputs)
 
-        # capture logging
-        self.trainer.logger_connector.cache_logged_metrics()
-
     def on_evaluation_batch_start(self, batch: Any, batch_idx: int, dataloader_idx: int) -> None:
+        self.trainer.logger_connector.on_batch_start()
+        # FIXME(@carmocca): missing hook?
+        # self.trainer.call_hook('on_batch_start')
```

Comment:
it's missing on purpose, I thought we decided not to run on_batch_start/end regardless of train/val/predict. It's only for training.

cc @ananthsub

Reply:
Where did we discuss this? 😂 
It's weird then because we do run `on_epoch_{start,end}` for them.

This is unrelated to the PR though, can remove the FIXME and address it again in #7738

Reply:
I initially suggested it in this issue a long time ago: https://github.com/PyTorchLightning/pytorch-lightning/issues/1440
Then later there was in a slack message thread. @ananthsub had an argument against, which I don't remember.